src/Arcus.WebApi.Security/Authorization/Jwt/JwtTokenReader.cs

```diff
@@ -17,10 +20,24 @@ public class JwtTokenReader : IJwtTokenReader
             "https://login.microsoftonline.com/common/v2.0/.well-known/openid-configuration";
 
         private readonly string _applicationId;
+        private readonly ClaimCheck _claimCheck;
         private readonly TokenValidationParameters _tokenValidationParameters;
         private readonly JwtSecurityTokenHandler _handler = new JwtSecurityTokenHandler();
         private readonly ConfigurationManager<OpenIdConnectConfiguration> _configManager;
 
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <remarks>Uses Microsoft OpenId connect discovery endpoint</remarks>
+        /// <param name="claimCheck">Custom claims key-value pair to validate against</param>
+        public JwtTokenReader(ClaimCheck claimCheck)
```

Comment:
Is it sufficient to only allow one `ClaimCheck` or should we allow to have multiple ?
Maybe use a constructor that looks like this:

```csharp
public JwtTokenReader( params ClaimCheck[] claimChecks)
```

Reply:
I want to leave a comment on line 45, but I'm unable to do so.

Should the constructor that takes the `applicationId` parameter call the new constructor ?  So in fact, the existing ctor can create a ClaimCheck instance for the `aud` claim, and use the `applicationId` as value ?

We can then also get rid of the `_applicationId` member.

Reply:
I was once told that we should limit the 'special' features in Arcus (was about `params`) so while I'm also a fan of `params`, maybe we should pass in an `IEnumerable<>` instead.

Reply:
It's not that we limit it because it's "special" but more how we consume it, I prefer a list over arrays as they are easier to use but if you both think params is better then that's fine for me.

Reply:
Yeah, sorry, my bad, 😅 , in this case, in a constructor, I don't think `params` is something that's helpful. It's more something you see in methods I think. (although I'm guilty of using it in ctors).